tests/google_service_test.py

```diff
@@ -0,0 +1,59 @@
+import os
+import requests
+from base_app.base import app
+from base_app.google_service import GoogleService
+from dotenv import load_dotenv
+load_dotenv()
+
+def test_can_initialize():
+    service = GoogleService()
+    assert isinstance(service, GoogleService)
+
+def test_google_api_returns_successful_response():
+    GOOGLE_API_KEY = os.getenv('GOOGLE_API_KEY')
+    address = '1331 17th St LL100, Denver, CO'
+    params = {'key': GOOGLE_API_KEY, 'address': f'{address}' }
+    service = GoogleService()
+    response = service.connection(params)
+    status = response.status_code
+    assert response.status_code == 200
+
+def test_google_api_returns_coordinates_with_street_address():
+    GOOGLE_API_KEY = os.getenv('GOOGLE_API_KEY')
+    address = '1331 17th St LL100, Denver, CO'
+    params = {'key': GOOGLE_API_KEY, 'address': f'{address}' }
+    service = GoogleService()
+    response = service.get_coordinates(params)
+    assert response == { 'latitude': 39.7508006, 'longitude': -104.9965947 }
+
+def test_google_api_returns_coordinates_with_city_and_state():
+    GOOGLE_API_KEY = os.getenv('GOOGLE_API_KEY')
+    address = 'Denver, CO'
+    params = {'key': GOOGLE_API_KEY, 'address': f'{address}' }
+    service = GoogleService()
+    response = service.get_coordinates(params)
+    assert response == { 'latitude': 39.7392358, 'longitude': -104.990251 }
+
+def test_google_api_returns_coordinates_with_only_city():
+    GOOGLE_API_KEY = os.getenv('GOOGLE_API_KEY')
+    address = 'Denver'
+    params = {'key': GOOGLE_API_KEY, 'address': f'{address}' }
+    service = GoogleService()
+    response = service.get_coordinates(params)
+    assert response == { 'latitude': 39.7392358, 'longitude': -104.990251 }
+
+def test_google_api_returns_coordinates_with_only_state():
+    GOOGLE_API_KEY = os.getenv('GOOGLE_API_KEY')
+    address = 'Colorado'
+    params = {'key': GOOGLE_API_KEY, 'address': f'{address}' }
+    service = GoogleService()
+    response = service.get_coordinates(params)
+    assert response == { 'latitude': 39.5500507, 'longitude': -105.7820674 }
+
+def test_google_api_returns_error_if_invalid_address_submitted():
```

Comment:
I like that you accounted for this sad path, but how does Google's API service determine what city/state to use for non-unique street addresses?

Reply:
Good question - FE is aware that we'd like the address autocomplete on the form, but it will be a stretch goal for them. I've looked into seeing if there's a way to do that on the backend, but I'm not seeing any good alternatives. Kim has changed the form on the front end to say "Street Address, City, and State" in the field to help prompt users. Aside from that, Google returns "partial matches" and will send back what it thinks is correct. It looks like that is determined based on locality or what it thinks the best answer to send back is.